pure-wasm/src/lib.rs

```diff
@@ -235,6 +273,62 @@ pub extern "C" fn check_enabled(engine_ptr: i32, message_ptr: i32, message_len:
     }
 }
 
+#[unsafe(no_mangle)]
+pub extern "C" fn check_variant(engine_ptr: i32, message_ptr: i32, message_len: i32) -> u64 {
```

Comment:
I think there's a harvested API making itself clear in this file but this is probably okay for this PR